### PR TITLE
Add nutanix-specific customizations for assorted  assets

### PR DESCRIPTION
**Pull request by thunderboltsid**
Add nutanix platform customizations in:
  - pkg/asset/cluster
  - pkg/asset/ignition
  - pkg/asset/quota
  - pkg/asset/rhcos

### CODE REVIEW

File: pkg/asset/rhcos/image.go [hunk 1]
```diff
@@ -163,6 +164,17 @@ func osImage(config *types.InstallConfig) (string, error) {
 		return osimage, nil
 	case none.Name:
 		return "", nil
+	case nutanix.Name:
+		op := config.Platform.Nutanix
```
Comment (staebler) [inline pkg/asset/rhcos/image.go:168]: What does `op` stand for?

File: pkg/asset/rhcos/image.go [hunk 2]
```diff
@@ -163,6 +164,17 @@ func osImage(config *types.InstallConfig) (string, error) {
...
+		if op != nil {
+			if oi := op.ClusterOSImage; oi != "" {
+				return oi, nil
+			}
+		}
+		if a, ok := streamArch.Artifacts["nutanix"]; ok {
+			return rhcos.FindArtifactURL(a)
+		}
+		return "", fmt.Errorf("%s: No Nutanix build found", st.FormatPrefix(archName))
```
Comment (staebler) [inline pkg/asset/rhcos/image.go:177]: The name of the artifact in the stream is "nutanix".
```suggestion
		return "", fmt.Errorf("%s: No nutanix build found", st.FormatPrefix(archName))
```

File: pkg/asset/rhcos/image.go (hunk 1 above)
Comment (thunderboltsid) [reply pkg/asset/rhcos/image.go:168]: openshift platform. Simplified the conditional to avoid temp variables.

**Review by staebler — COMMENTED** [review]
/approve

**Pull request merged**